tempodb/pool/pool.go

```diff
@@ -118,16 +118,18 @@ func (p *Pool) RunJobs(ctx context.Context, payloads []interface{}, fn JobFunc)
 	// wait for all jobs to finish
 	wg.Wait()
 
-	// see if anything ended up in the results channel
-	var msg []byte
-	select {
-	case msg = <-resultsCh:
-	default:
+	// close resultsCh
+	close(resultsCh)
+
+	// read all from results channel
+	var msg [][]byte
+	for res := range resultsCh {
+		msg = append(msg, res)
 	}
 
 	// ignore err if msg != nil.  otherwise errors like "context cancelled"
 	//  will take precedence over the err
-	if msg != nil {
+	if len(msg) > 0 {
```

Comment:
i don't think this is true anymore.  we were taking into account the context cancelled errors b/c we were purposefully cancelling context if a single job completed successfully.   now that we are allowing all jobs to finish I think we should let errors take precedence.  WCGW? :)

Reply:
Yeah since we are no longer cancelling contexts on the individual jobs, I agree `err` should be given precedence here.